website/project/views/contributor.py

```diff
@@ -867,21 +876,31 @@ def claim_user_form(auth, **kwargs):
 def _add_related_claimed_tag_to_user(pid, user):
     """
     Adds claimed tag to incoming users, depending on whether the resource has related source tags
-    :param pid: guid of either the node or the preprint
+    :param pid: guid of either the node or the preprint or the registration
     :param user: the claiming user
     """
     node = AbstractNode.load(pid)
     preprint = Preprint.load(pid)
+    registration = Registration.load(pid)
     osf_claimed_tag, created = Tag.all_tags.get_or_create(name=provider_claimed_tag('osf'), system=True)
-    if node:
-        node_source_tags = node.all_tags.filter(name__icontains='source:', system=True)
-        if node_source_tags.exists():
-            for tag in node_source_tags:
-                claimed_tag, created = Tag.all_tags.get_or_create(name=NODE_SOURCE_TAG_CLAIMED_TAG_RELATION[tag.name],
-                                                                  system=True)
-                user.add_system_tag(claimed_tag)
+    if registration:
+        registration_id = registration.provider._id
+        registration_claimed_tag, created = Tag.all_tags.get_or_create(name=provider_claimed_tag(registration_id, 'registry'), system=True)
+        user.add_system_tag(registration_claimed_tag)
+    elif node:
+        if node.is_collected:
```

Comment:
Drive by: I'm not sure if the logic on the claimed is quite right. I think for claiming it's more important that we catch the OSF4M case or the two prereg campaigns, because those would likely be where the user was added from. To clarify, OSF4M and the prereg campaign tags mean that the node was created as part of those actions, rather than collections, which is entirely separate from the creation. So a user claiming a collected node that started as OSF4M, for example, is more correctly claimed from OSF4M. Does that make sense?

If we capture all of them, that's probably fine, but if we're just getting a subset, I'd lean towards the tags that are currently in your else block.

Reply:
Sounds good - Made changes for prioritization of the source tags for nodes in collections as well